### PR TITLE
Fix edit command

### PR DESCRIPTION
**Pull request by HongRay**
Fix edit command 

Bug fix for the edit command
1. Does not detect whether category exists: (FIX) Throws exception if does not exist.
2. When description is not provided, returns null (FIX) Throws an exception if description is not provided.
3. When tag not provided, deletes tag (FIX) Finds if tag present and if tag size = 0, throws an error. 
4. Tag can only edit 1 tag. (FIX) If a person inputs more than 1 tag, the subsequent tags will be deleted. (If want more than 1 tag, concatenate tags prefix) 
5. Cannot change the category name (will issue this as an added additional feature)

### CODE REVIEW

**Comment by Cedricaca** [comment]
Looks good bro

I really like how you changed the code while utilising your own writing style, and with no checkstyle errors

I think you have been generating very good code also,

hopefully this comment gets picked up by the algorithms

**Comment by Xuyan0518** [comment]
Looks bro. Thanks for helping to fix the bugs mentioned above. I have checked and they work as promised. Good job!

I wish to be able to code like you do.

**Comment by ChenXy128** [comment]
LGTM! Just add in some JavaDoc and it will be a lot more clearer.

Code Quality is up to standard.

Easy to read as well

**Comment by NusMinato** [comment]
Amazing job bro! I love the way you fixed bugs.
The problem is very well analysed.
The code looks clear and easy to understand, with no redundancies.
Well done!

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -83,16 +83,20 @@ public CommandResult execute(Model model) throws CommandException {
...
+            editPersonDescriptor) throws CommandException {
         assert personToEdit != null;
         if (editPersonDescriptor.getCategory() != null) {
             Entry personToEditEntry = personToEdit.getEntry(editPersonDescriptor.getCategory());
-            personToEditEntry.setDescription(editPersonDescriptor.getDescription());
+            if (personToEditEntry == null) {
+                throw new CommandException(MESSAGE_CATEGORY_DOESNT_EXIST);
+            } else {
+                personToEditEntry.setDescription(editPersonDescriptor.getDescription());
+            }
```
Comment (NusMinato) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:99]: Well done! Solved the bug of non-existent category perfectly.

**Review by NusMinato — COMMENTED** [review]
Amazing job bro! I love the way you fixed bugs.
The problem is very well analysed.
The code looks clear and easy to understand, with no redundancies.
Well done!

**Pull request merged**